application-observability/stock-snap-py/app.py

```
@@ -108,6 +114,35 @@ def get_stock_price():
             random_sleep()  # Simulate fetching delay
             price = stock_prices[symbol]
 
+            # Alternatively, retrieve the stock price from your Snowflake account
+            # The connector library is equipped with trace propagation
+            
+            # SPCS stores your account information in the SPCS instance environment
+            SNOWFLAKE_ACCOUNT = os.getenv('SNOWFLAKE_ACCOUNT')
+            SNOWFLAKE_HOST = os.getenv('SNOWFLAKE_HOST')
+            
+            conn = snowflake.connector.connect(
+                host = SNOWFLAKE_HOST,
+                account = SNOWFLAKE_ACCOUNT,
+                token = get_login_token(),
+                authenticator = "oauth"
+            )
+
+            try:
+                cur = conn.cursor()
+                cur.execute("USE DATABASE STOCKS;")
+                cur.execute("USE SCHEMA STOCK_PRICES;")
+                cur.execute(f"SELECT price FROM stock_prices WHERE symbol = ?", (symbol,))
+                price = cur.fetchone()[0]
+                conn.commit()
+
+            except Exception as e:
+                print(str(e))
```

Just raise this exception instead of wrapping in `print`.